packages/wallet/frontend/src/pages/grants/[grantId].tsx

```diff
@@ -0,0 +1,135 @@
+
+const GrantPage: NextPageWithLayout<GrantPageProps> = ({ grant }) => {
+  const [openDialog, closeDialog] = useDialog()
+  const router = useRouter()
+
+  const handleRevokeConfirmation = async (id: string) => {
+    const response = await grantsService.delete(id)
+    if (response.success) {
+      openDialog(
+        <SuccessDialog onClose={closeDialog} content={response.message} />
+      )
+      router.reload()
+    } else {
+      openDialog(
+        <ErrorDialog onClose={closeDialog} content={response.message} />
+      )
+    }
+  }
+
+  return (
+    <>
+      <div className="flex flex-col items-start md:flex-col">
+        <PageHeader title="Grant details" />
+        <div className="my-16 flex flex-col space-y-5 text-xl text-green">
+          <div>
+            <span className="font-semibold">Client:</span>
+            <span className="font-light"> {grant.client}</span>
+          </div>
+          <div>
+            <span className="font-semibold">Created at: </span>
+            <span className="font-light">{grant.createdAt}</span>
+          </div>
+          <div>
+            <span className="font-semibold">Payment Pointer access: </span>
+            <span className="font-light">{grant.access.identifier}</span>
+          </div>
+          <div className="flex items-center">
+            <span className="mr-4 font-semibold">State:</span>
+            <Badge
+              intent={getStatusBadgeIntent(grant.state)}
+              size="md"
+              text={grant.state}
+            />
+          </div>
+        </div>
+        {grant.state !== 'REVOKED' && (
+          <Button
+            intent="secondary"
+            aria-label="revoke"
+            onClick={() => {
+              openDialog(
+                <ConfirmationDialog
+                  confirmText="Revoke Grant"
+                  warningText="Are you sure you want to revoke this grant?"
+                  onConfirm={() => handleRevokeConfirmation(grant.id)}
+                  onClose={closeDialog}
+                />
+              )
+            }}
+          >
+            Revoke Grant
+          </Button>
+        )}
+      </div>
+      <Image
+        className="mt-28 object-cover"
+        src="/grants.webp"
+        alt="Grants"
+        quality={100}
+        width={700}
+        height={200}
+      />
+    </>
+  )
+}
+
+const querySchema = z.object({
+  grantId: z.string().uuid()
+})
+
+export const getServerSideProps: GetServerSideProps<{
+  grant: Grant
+}> = async (ctx) => {
+  const result = querySchema.safeParse(ctx.query)
+
+  if (!result.success) {
+    return {
+      notFound: true
+    }
+  }
+  const grantResponse = await grantsService.get(
+    result.data.grantId,
+    ctx.req.headers.cookie
+  )
+
+  if (!grantResponse.success || !grantResponse.data) {
+    return {
+      notFound: true
+    }
+  }
+
+  grantResponse.data.createdAt = formatDate(grantResponse.data.createdAt)
```

Comment:
This should be moved to the client.

Reply:
After talks with Radu we will leave this as it is for now